core/domain/state_domain.py

```diff
@@ -1367,7 +1367,8 @@ class State(object):
 
     def __init__(
             self, content, param_changes, interaction, recorded_voiceovers,
-            written_translations, classifier_model_id=None):
+            written_translations, solicit_answer_details=False,
```

Comment:
As currently whenever a state will be created it will have its solicit_answer_details to be false by default. So when the creator will create a new state its solicit_answer_details will be false, and later on, he can enable it i.e. make it true by marking the checkbox(which I will implement in the next PR).

Reply:
I'll repeat my previous comment -- don't have a default arg here. Callers of this constructor should, in general, specify all args explicitly.

Reply:
Done